@types/index.test-d.ts

```diff
@@ -62,31 +62,24 @@ async function run() {
 	}
 
 	// export *
-	const wildRes = await _fetch('https://google.com');
-	expectType<boolean>(wildRes.ok);
-	expectType<number>(wildRes.size);
-	expectType<number>(wildRes.status);
-	expectType<string>(wildRes.statusText);
-	expectType<() => Response>(wildRes.clone);
-
-	// export = require
-	const reqRes = await __fetch('https://google.com');
-	expectType<boolean>(reqRes.ok);
-	expectType<number>(reqRes.size);
-	expectType<number>(reqRes.status);
-	expectType<string>(reqRes.statusText);
-	expectType<() => Response>(reqRes.clone);
+	const wildResponse = await _fetch('https://google.com');
+	expectType<boolean>(wildResponse.ok);
+	expectType<number>(wildResponse.size);
+	expectType<number>(wildResponse.status);
+	expectType<string>(wildResponse.statusText);
+	expectType<() => Response>(wildResponse.clone);
```

Comment:
This part fails the tests, meaning TypeScript users cannot import node-fetch like so:

```ts
import * as fetch from 'node-fetch';
```

I tried to fix this issue, but without success. I also wonder whether we really need to test this - are there any benefits from using the aforementioned `import..as` style than just doing:

```ts
import fetch from 'node-fetch';
```

?

Reply:
this seems to be correct. `import * as ...` is importing the module, which is **always** an object.

if you want to use `import * as fetch` you would need to run `fetch.default()` in order to use it.

I believe the reason `import * as fetch from 'node-fetch';` worked in typescript with commonjs is because of module interop: https://www.typescriptlang.org/tsconfig#esModuleInterop

Reply:
@dnalborczyk Thanks! This explains everything :smile: